e2e/src/app.po.ts

```diff
@@ -2,10 +2,81 @@ import { browser, by, element } from 'protractor';
 
 export class AppPage {
   navigateTo() {
-    return browser.get(browser.baseUrl) as Promise<any>;
+    return browser.get('/');
   }
 
-  getTitleText() {
-    return element(by.css('app-root .content span')).getText() as Promise<string>;
+  openDialog() {
+    this.getOpenDialogButton().click();
+    browser.wait(this.getModalDialog().isDisplayed, 3000);
+  }
+
+  getActionItemsTable() {
+    return element(by.css('.tableContainer'));
+  }
+
+  getActionItems() {
+    return element.all(by.css('.singleItem'));
+  }
+
+  getOpenDialogButton() {
+    return element(by.css('.addItem'));
+  }
+
+  getModalDialog() {
+    return element(by.css('.dialogModal'));
+  }
+
+  getCreateActionItemButton() {
+    return element(by.css('.create'));
+  }
+
+  getCloseDialogButton() {
+    return element(by.css('.close'));
+  }
+
+  getMatSelect() {
+    return element(by.tagName('mat-select'));
+  }
+
+  getMatOption() {
+    return element(by.cssContainingText('mat-option', 'CASD Wilson & Lamberton Middle Schools'));
+  }
+
+  getItemNameInput() {
+    return element(by.css('.nameInput'));
+  }
+
+  getProjectInput() {
+    return element(by.css('.projectInput'));
+  }
+
+  getDateInput() {
+    return element(by.css('.dateInput'));
+  }
+
+  passActionItemName() {
+    return this.getItemNameInput().sendKeys(this.itemUUID());
+  }
+
+  passDueDate() {
+    return this.getDateInput().sendKeys('2019/11/15');
+  }
+
+  itemUUID() {
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, c => {
+      const r = (Math.random() * 16) | 0,
+        v = c === 'x' ? r : (r & 0x3) | 0x8;
+      return v.toString(16);
+    });
+  }
+
+  getItemNameInputId() {
+    const name = this.getItemNameInput();
+    name.id = this.itemUUID();
+    return name.id;
+  }
+
+  getNewItem() {
+    return element(by.cssContainingText(this.getItemNameInputId(), this.itemUUID()));
```

Comment:
**CONCERN:** Isn't this test false positive?  `this.itemUUID()` will always generate a new unique name, so in my opinion this should never be true.